sklearn/metrics/tests/test_ranking.py

```diff
@@ -123,6 +129,40 @@ def _average_precision(y_true, y_score):
     return score / n_pos
 
 
+def _average_precision_slow(y_true, y_score):
+    """A second alternative implementation of average precision that closely
+    follows the Wikipedia article's definition (see References). This should
+    give identical results as `average_precision_score` for all inputs.
+
+    References
+    ----------
+    .. [1] `Wikipedia entry for the Average precision
+       <http://en.wikipedia.org/wiki/Average_precision>`_
+    """
+    precision, recall, threshold = precision_recall_curve(y_true, y_score)
+    precision = list(reversed(precision))
+    recall = list(reversed(recall))
+    average_precision = 0
+    for i in range(1, len(precision)):
+        average_precision += precision[i] * (recall[i] - recall[i - 1])
+    return average_precision
+
+
+def _interpolated_average_precision_slow(y_true, y_score):
+    """A second implementation for the eleven-point interpolated average
+    precision used by Pascal VOC. This should produce identical results to
+    average_precision_score with `interpolation='eleven_point'`.
+    """
+    precision, recall, _ = precision_recall_curve(y_true, y_score)
+    precision = list(reversed(precision))
+    recall = list(reversed(recall))
+    eleven_precisions = list()
+    for threshold in np.arange(0, 1.1, 0.1):
```

Comment:
I'll leave this change until we work out how we want to implement 11-point average precision.